test/e2e/restart.go

```diff
@@ -0,0 +1,367 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	// How long each node is given during a process that restarts all nodes
+	// before the test is considered failed. (Note that the total time to
+	// restart all nodes will be this number times the nubmer of nodes.)
+	restartPerNodeTimeout = 5 * time.Minute
```

Comment:
1. By default, devs and jenkins launch clusters with only two nodes (a la [`cluster/gce/config-test.sh`](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/gce/config-test.sh#L23)), so this will happen. Furthermore, configuring to only rolling-update to some of the nodes would involve passing an `--auto-pause-after-instances` flag, which would change the test: we would need to look for "PAUSED" state instead of "ROLLED_OUT", but it's unclear how "PAUSED" is different than "CANCELLED", and which one will be applied if the rolling update fails. (I can't easily tell because I haven't gotten one to fail yet.) Anyway, tl;dr is we have two-node clusters, so this should only apply to two nodes.

2. I vote we punt on parallel right now. We can parameterize this and set up a dedicated Jenkins job if we decide to investigate this later. (I'm curious: why is it unlikely to be successful? If I understand correctly, all necessary state to reschedule all pods is on the master, so blowing away all nodes at once shouldn't cause any problems, right?)

3. Shouldn't be necessary (a la 1).

Re: soak tests, the only one I saw that disabled other tests (like reboot) was "kubernetes-soak-continuous-e2e", so I disabled this test on that already. Are there other Jenkins jobs I should do this to?

Re: large clusters, are there any Jenkins jobs that run > 2 size clusters for which I should also disable this test?